### ISSUE TITLE
Error message scoped to `git`

### ISSUE THREAD

**Issue by dleve123** [thoughtbot/gitsh#240]
A "picture" speaks a thousand words:

```bash
$ gitsh
extract-class master@ add
Nothing specified, nothing added.
Maybe you wanted to say 'git add .'?
```

To those that are unfamiliar with `git` and `gitsh`, this error message might be somewhat confusing as `git add .` leads to an error message.

If the maintainers are interesting in fixing this issue, I'm more than happy to help out.

**Comment by georgebrock** [comment]
I agree this is confusing, but I'm not sure how we'd cleanly fix it. I've considered a global substitution on error output, but Git has so many commands with so many potential error messages that I suspect we'd end up with false positives, making some messages worse and not better.

Any thoughts on you'd approach it?

**Comment by dleve123** [comment]
One idea that comes to mind is prefixing the Git error message with some sort of minimal text that indicates that this is a `git` and not strictly a `gitsh` error. Then it would be implied (maybe) that code output shouldn't be copied and pasted wholesale.

The output could look something like this:

```bash
$ gitsh
extract-class master@ add
[git-level error]
Nothing specified, nothing added.
Maybe you wanted to say 'git add .'?
```

What do you think? Easier to extend the error than to modify it methinks.

**Comment by georgebrock** [comment]
Another example in non-error output from `git status`:

```
…
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)
…
```

**Comment by dleve123** [comment]
@georgebrock 

Another solution that comes to mind is to simply ignore the token "git" from expressions that start with "git". So,

```sh
$ gitsh
extract-class master@ git add
```

would behave in an identical fashion to:

```sh
$ gitsh
extract-class master@ add
```

Note, `git git` when executed in a normal Bash shell outputs an error, so I don't think there would be any mucking with Git functionality here.

How would you feel about a PR that implements this?

**Comment by georgebrock** [comment]
@dleve123 gitsh already supports this behaviour, but only when Git's autocorrect feature is enabled (see #182 for background)

**Comment by dleve123** [comment]
@georgebrock Oh snap, well that changes everything then.

Let me know what direction you want to take this in. I'm down to pursue stripping "git" from output, but perhaps it would be best / most minimal to just compel users to turn on Git's autocorrect in the project's README? While such a README change would be slightly redundant with the doc diff introduced at https://github.com/thoughtbot/gitsh/pull/182, it would surely be more approachable to a new gitsh user.

**Comment by georgebrock** [comment]
@dleve123 I'd personally prefer to keep the README for high-level overview and installation stuff, and keep the functional detail in the manual page. Maybe people don't read the manpage, though.
